Swiften/Client/Client.cpp

```diff
@@ -148,13 +154,27 @@ void Client::handleConnected() {
     discoManager->handleConnected();
 }
 
+void Client::requestFeatures() {
+    auto discoInfo = std::make_shared<DiscoInfo>();
+    auto request = std::make_shared<GenericRequest<DiscoInfo>>(IQ::Get, JID(), discoInfo, getIQRouter());
+    request->onResponse.connect(boost::bind(&Client::handleFeaturesReceived, this, _1, _2));
+    request->send();
+}
+
+void Client::handleFeaturesReceived(DiscoInfo::ref payload, ErrorPayload::ref /*error*/) {
+    if (payload && payload->hasFeature(DiscoInfo::MIXFeature)) {
+        mixSupport = true;
+        mixRegistry = std::make_unique<MIXRegistry>(getJID(), getIQRouter(), getRoster());
```

Comment:
Always having a mixRegistry, but it being empty when not supported, probably makes more sense, doesn't it?